app/src/main/java/com/epfl/drawyourpath/mainpage/MainActivity.kt

```diff
@@ -113,84 +82,19 @@ class MainActivity : AppCompatActivity() {
-            profileImageButton.setImageBitmap(it.profilePhoto(resources))
-        }
-        drawerLayout = findViewById(R.id.drawerLayout)
-        // Set a listener to open the drawer menu (we might want it on the right)
-        profileImageButton.setOnClickListener {
-            drawerLayout.open()
-        }
-    }
-
-    private fun setupDrawerNavigationView() {
-        val drawerNavigationView: NavigationView = findViewById(R.id.navigationView)
-        val header = drawerNavigationView.getHeaderView(0)
-        userCached.getUser().observe(this) {
-            header.findViewById<TextView>(R.id.header_username).text = it.username
-            header.findViewById<TextView>(R.id.header_email).text = it.emailAddress
-        }
-        // Handle the items in the drawer menu
-        drawerNavigationView.setNavigationItemSelectedListener { menuItem ->
-            when (menuItem.itemId) {
-                // Display profile fragment
-                R.id.profile_menu_item -> replaceFragment<ProfileFragment>(
-                    bundleOf(
-                        PROFILE_USER_ID_KEY to userCached.getUserId(),
-                    ),
-                )
-
-                // Display stats fragment
-                R.id.stats_menu_item -> replaceFragment<StatsFragment>()
-
-                // Display challenge fragment
-                R.id.challenge_menu_item -> replaceFragment<ChallengeFragment>()
-
-                // Display settings fragment
-
-                R.id.preferences_menu_item -> replaceFragment<PreferencesFragment>()
-            }
-            drawerLayout.close()
-            true
-        }
-    }
-
-    private fun setupBottomNavigationView() {
-        bottomNavigationView = findViewById(R.id.bottom_navigation)
-        // Listener that handles the items in the bottom menu
-        bottomNavigationView.setOnItemSelectedListener { menuItem ->
-            when (menuItem.itemId) {
-                // Display community fragment
-                R.id.community_menu_item -> replaceFragment<CommunityFragment>()
-
-                // Display friends fragment
-                R.id.friends_menu_item -> replaceFragment<FriendsFragment>()
-
-                // Display drawing fragment
-                R.id.draw_menu_item -> replaceFragment<DrawMenuFragment>()
-
-                // Display history fragment
-                R.id.history_menu_item -> replaceFragment<HistoryFragment>()
-
-                // Display chat fragment
-                R.id.chat_menu_item -> replaceFragment<ChatFragment>()
-            }
-            true
-        }
-    }
-
     private inline fun <reified F : Fragment> replaceFragment(args: Bundle? = null) {
         supportFragmentManager.commit {
             setReorderingAllowed(true)
-            replace(R.id.fragmentContainerView, F::class.java, args)
+            replace(R.id.main_fragment_container_view, F::class.java, args)
+        }
+    }
+
+    private fun setupFragment() {
```

Comment:
Maybe add a comment to explain how this function behave in function of his arguments